### PR TITLE
Shell error messages improvements

### PR DESCRIPTION
**Pull request by ret2libc**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

* When a command does not exist it shows a better error message making clear the command does not exist. This works only if the whole commands sub-tree is handled by newshell, otherwise things are redirected to the old-style handlers, which may or may not make it clear a command does not exist.
* When a user passes the wrong number of arguments the help for that command is automatically shown to the user
* When a user runs things like `wv5? <arg>` which in general might be a command with an arg, we show the error `Command 'wv5?' does not exist. Did you want to see the help? Try "wv5?" without any argument.`

### CODE REVIEW

**Comment by Maijin** [comment]
"[...]because not existent" is not English, you may want to choose one of the following:

* [...] because non-existent
* [...] because it doesn't exist (I prefer this one)

Same with NOTEXISTING, I'd use NONEXISTING or something used in other similar projects (if that's a thing).

**Review by XVilka — APPROVED** [review]
Would be very helpful for users, indeed.

File: librz/core/cmd.c [hunk 1]
```diff
@@ -4635,6 +4635,23 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(arged_command) {
...
 		const char *cmdname = rz_cmd_parsed_args_cmd (pr_args);
 		eprintf ("Wrong number of arguments passed to `%s`, see its help with `%s?`\n", cmdname, cmdname);
+		RzCmdDesc *cd = rz_cmd_get_desc (state->core->rcmd, cmdname);
+		if (cd) {
+			char *cmdname_help = rz_str_newf ("%s?", cmdname);
+			if (!cmdname_help) {
+				goto err;
+			}
+			RzCmdParsedArgs *help_pra = rz_cmd_parsed_args_newcmd (cmdname_help);
+			char *help_msg = rz_cmd_get_help (state->core->rcmd, help_pra, true);
```
Comment (wargio) [inline librz/core/cmd.c:4648]: `help_msg` is not checked of its value. so if something goes wrong, it will print `(null)`

**Review by wargio — CHANGES_REQUESTED** [review]

File: librz/core/cmd.c (hunk 1 above)
Comment (wargio) [reply librz/core/cmd.c:4648]: is `help_pra` dynamically allocated? if yes then it needs to be freed, otherwise probably it should be a `const` in there

Comment (ret2libc) [reply librz/core/cmd.c:4648]: fixed

**Review by Maijin — APPROVED** [review]

**Pull request merged**